### DISCUSSION TITLE
Improve Vercel "tokens" page - Teach about tokens auto-removal, improve DX and allow for custom token lifetime

### DISCUSSION THREAD

**Discussion by Vadorequest** [vercel/vercel#5112] [Ideas] [answered]
Following discussion in https://github.com/vercel/vercel/discussions/5098

It would be great to do the following on https://vercel.com/account/tokens page:

1. **Explicit the fact tokens are being reset after 10 days of inactivity** (as mentioned in https://vercel.com/blog/changelog-august-2020)
This will help people who get their tokens deleted figure out why. Also, it'll educate devs who create new tokens and aren't aware of that until they run into it.

1. **Send a notification (email) when a token is automatically removed due to inactivity**
This will help resolve issues by avoiding the embarrassing "this worked fine but doesn't anymore and I don't know why".

1. **Allow to increase the duration of an inactive token lifetime from 10 days to up to 30 days**
Personally, I'd want all my CI/CD tokens to have an infinite lifespan if possible.
If it's not technically possible, something like 3 months would be much better than 10 days.
I went to holidays for 14 days and when I came back, all my CI/CD pipelines were broken because all Vercel tokens were deleted due to inactivity, it's not a nice welcome back gift at all. 😕 

1) and 2) are must-haves IMHO and should be taken care of quickly to avoid issues like https://github.com/vercel/vercel/discussions/5098 and bad DX.
3) will likely require a bit of work, but I believe there are many valid use cases where 10 days isn't recommended, even though I do believe it's a proper default.

**Comment by one000mph** [comment]
+1 to this issue. there's no notification when tokens expire. Adding tokens to my CI is tedious busywork and is frequently necessary with the current limitations. We are a small team and don't deploy super frequently so tokens are expiring on us a lot :(

**Comment by leo** [comment, marked as answer]
Thanks for your feedback! We'll take it into consideration when overhauling this page the next time.

For now, we're already working on an immediate fix for the token expiration situation by making sure that only tokens created by logging into the Dashboard are automatically expired. That should resolve several of your problems.

**Reply by one000mph** [reply]
what is the alternate way to create a token without using the dashboard? It doesn't look like they can be created via the CLI.

**Reply by Vadorequest** [reply]
@one000mph You misunderstood Leo, only tokens created **by logging into** the dashboard are now automatically expiring. Not those created through the dashboard.

I guess many tokens created by authenticating into the dashboard were left alive and growing, and that's why they implemented some auto-removal feature, which also deleted manually-created tokens by mistake, and why they recently changed that behaviour to only auto-remove those created automatically.

**Comment by ide** [comment]
A few months ago I wrote a small GitHub Actions workflow to keep a token active.

```yaml
name: Keep Vercel token active
on:
  schedule:
    - cron: '0 0 * * MON'
jobs:
  refresh:
    runs-on: ubuntu-latest
    steps:
    - name: Use Vercel token
      shell: bash
      env:
        VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
      run: npx vercel --token "$VERCEL_TOKEN" whoami
```

This keep tokens alive indefinitely as long as the repo exists. You could also use this for your dashboard tokens too if you don't want to keep getting logged out.
